app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt

```diff
@@ -35,4 +60,29 @@ class EnterOtpScreenController @Inject constructor(
     return events.ofType<EnterOtpBackClicked>()
         .map { { ui: Ui -> ui.goBack() } }
   }
+
+  private fun handleOtpSubmitted(events: Observable<UiEvent>): Observable<UiChange> {
+    return events.ofType<EnterOtpSubmitted>()
+        .filter { it.otp.length == OTP_LENGTH }
+        .flatMapSingle { userSession.loginWithOtp(it.otp) }
+        .map { loginResult ->
+          when (loginResult) {
+            is LoginResult.Success -> { ui: Ui ->
+              startSyncing()
+              ui.goBack()
+            }
+            is LoginResult.UnexpectedError -> { ui: Ui -> ui.showUnexpectedError() }
+            is LoginResult.ServerError -> { ui: Ui -> ui.showServerError(loginResult.error) }
+            is LoginResult.NetworkError -> { ui: Ui -> ui.showNetworkError() }
+          }
+        }
+  }
+
+  private fun startSyncing() {
+    syncScheduler.syncImmediately()
+        .subscribeOn(Schedulers.io())
+        // Swallow errors
```

Comment:
This comment isn't necessary. `onErrorComplete()` already says the same thing :)

Reply:
This was changed in the next commit. :P